### PR TITLE
Example of non-linear DAG

### PR DESCRIPTION
**Pull request by mariyadavydova**
This example shows an example of a batch which builds two test datasets, counts averages and sum of them, and prints out the result.

Upon writing this DAG I found several issues/missing pieces in batch mode:

* Building images for batch mode requires adding an image to live mode config and building from there.
* The bake logs need details, at least statuses of tasks and their outputs (without this information I have to check status and logs manually by copying job ID).
* 'project.yml' seems to be ignored (as 'flow.project_id' is resolved to 'demo_jobs' instead of 'neuro_flow_demo' both in live and batch modes).
* If a task job fails because it can't find a Docker image, the bake doesn't see that as a problem and doesn't stop.
* Inspect command doesn't show the ids of jobs, so it is hard to read the logs. 

There is a room for improvement even in this small example:

* We could cache dataset generation and calculating averages (imagine we build datasets of 1m numbers).
* Actions may allow us to avoid duplication (see dataset generation and avg calculation steps).

### CODE REVIEW

File: examples/demo-jobs/.neuro/demo.yml [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+      import numpy
+      dataset = numpy.load("/data/dataset.npy")
+      avg = numpy.average(dataset)
+      print("::set-output name=avg::", avg, sep='', flush=True)
+
+  - id: sum
+    needs: [first_avg, second_avg]
+    image: ${{ images.numpy.ref }}
+    python: |
+      print("::set-output name=sum::", ${{ needs.first_avg.outputs.avg }} + ${{ needs.second_avg.outputs.avg }}, sep='', flush=True)
```
Comment (asvetlov) [inline examples/demo-jobs/.neuro/demo.yml:80]:
```suggestion
      result = ${{ needs.first_avg.outputs.avg }} + ${{ needs.second_avg.outputs.avg }}
      print(f"::set-output name=sum::{result}")
```

**Review by asvetlov — COMMENTED** [review]
we can convert four `{first,second}_dataset` and `{first,second}_avg` steps into two `dataset` and `avg` steps, both are with matrix.
I believe this way is more idiomatic plus it has educational value.

File: examples/demo-jobs/.neuro/demo.yml [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+images:
+  numpy:
+    ref: image:${{ flow.project_id }}:v1.0
+    dockerfile: ${{ flow.workspace }}/Dockerfile-batch
+    context: ${{ flow.workspace }}
+
+volumes:
+  dataset_1:
+    remote: storage:.data/${{ flow.id }}/${{ args.size_first }}-${{ args.random_seed }}
```
[the remark this answers is not in the document]
Comment (mariyadavydova) [reply examples/demo-jobs/.neuro/demo.yml]: Fixed

**Comment by mariyadavydova** [comment]
@asvetlov Please take a look at #59, where I tried to apply matrix strategy with args to this example.

**Comment by mariyadavydova** [comment]
Issues #60, #61, #62, #63, #64 are created to address the feedback in this PR.

**Comment by mariyadavydova** [comment]
@asvetlov Can I merge this PR? I thought you already did and accidentally found out that it is still open :)

**Comment by asvetlov** [comment]
You can but I had a thought that you wanted to apply a matrix to dataset and avg calculations.

**Comment by mariyadavydova** [comment]
It is applied in different PR (#49). 
Well, OK, I will review both versions of the config and probably create a new PR instead of these two, because there some new features were added since I wrote this PR.

**Comment by mariyadavydova** [comment]
I close this as we already have a lot of good examples.

**Pull request closed without merging**